### PR TITLE
Notes on experiment with Apache Arrow, and some notes with interesting links

### PR DESCRIPTION
**Pull request by stvoutsin**
The branch naming is wrong, this doesn't address issue #60 . It's probably most closely related to issue #103 
In any case, creating a pull request, as this just contains some brief notes on a test with Apache Arrow and some interesting links addressing Spark setups.

### CODE REVIEW

File: notes/stv/20200429-apache-arrow.txt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+Attempting non-optimization as 'spark.sql.execution.arrow.fallback.enabled' is set to true.
+
+
+
+# After repeating the cell run, the error disappears..(!?)
+
+
+
+# Selecting the above gaia query dataframe into a pandas object
+# user@worker
```
Comment (Zarquan) [inline notes/stv/20200429-apache-arrow.txt:113]: > user@worker
On a worker node, or in Zeppelin ?

File: notes/stv/20200429-apache-arrow.txt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+# -------------------------------
+# Let's Try running the following Zeppelin cell from the notebook example:
+
+%spark.pyspark
+#df = pd.DataFrame(df, columns=df.columns)
+#print(type(df))
+#df.head()
+# ... doesn't work for me for the system install of pandas. Try this:
```
Comment (Zarquan) [inline notes/stv/20200429-apache-arrow.txt:121]: Do we know why the commented out code doesn't work ?

File: notes/stv/20200429-apache-arrow.txt [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+pandas_df = df.select("*").toPandas()
+print(type(pandas_df))
```
Comment (Zarquan) [inline notes/stv/20200429-apache-arrow.txt:123]: What results do you get ?

File: notes/stv/20200429-apache-arrow.txt [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+pandas_df.head()
+
+
```
Comment (Zarquan) [inline notes/stv/20200429-apache-arrow.txt:126]: So in summary -
1.  Ran the SQL query at line 36-42 to create a Spark data frame, and got an exception when trying to convert the Spark data frame to a Pandas data frame.
2. Install Apache Arrow on _all_ of the worker nodes
3.  Ran the **same** SQL query at line 36-42 to create a Spark data frame, and converting the Spark data frame to a Pandas data frame now works ?

File: notes/stv/20200429-spark-tips-ML.txt [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+ |-- households: string (nullable = true)
+ |-- housingMedianAge: string (nullable = true)
+ |-- latitude: string (nullable = true)
+ |-- longitude: string (nullable = true)
+ |-- medianHouseValue: string (nullable = true)
+ |-- medianIncome: string (nullable = true)
+ |-- population: string (nullable = true)
+ |-- totalBedRooms: string (nullable = true)
+ |-- totalRooms: string (nullable = true)
+
```
Comment (Zarquan) [inline notes/stv/20200429-spark-tips-ML.txt:40]: Is this this data set ?
https://www.datacamp.com/community/tutorials/apache-spark-tutorial-machine-learning#loading-in-your-data
https://www.dcc.fc.up.pt/~ltorgo/Regression/cal_housing.html

File: notes/stv/20200429-spark-tips-ML.txt [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#  
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+
+
```
Comment (Zarquan) [inline notes/stv/20200429-spark-tips-ML.txt:22]: Is this a copy of this tutorial ?
https://www.datacamp.com/community/tutorials/apache-spark-tutorial-machine-learning

File: notes/stv/20200429-spark-tips-ML.txt [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+    return sqrt(sum([x**2 for x in (point - center)]))
+
+WSSSE = parsedData.map(lambda point: error(point)).reduce(lambda x, y: x + y)
+print("Within Set Sum of Squared Error = " + str(WSSSE))
+
+# Save and load model
+clusters.save(sc, "target/org/apache/spark/PythonKMeansExample/KMeansModel")
+sameModel = KMeansModel.load(sc, "target/org/apache/spark/PythonKMeansExample/KMeansModel
+
+
```
Comment (Zarquan) [inline notes/stv/20200429-spark-tips-ML.txt:170]: What results do you get ?
What does it mean ?
Can we use this as an example ?

File: notes/stv/20200429-spark-tips-ML.txt (hunk 6 above)
Comment (stvoutsin) [reply notes/stv/20200429-spark-tips-ML.txt:22]: Not a copy but that is the tutorial I followed (there's reference to that link in the notes as well) in the process of trying to learn a bit about ML and how to work with Dataframes and copied bits that I found useful and wanted to have . There is also a snippet from https://spark.apache.org/docs/latest/mllib-clustering.html

**Pull request merged**